### PR TITLE
adding new pending file interface + logic related to it

### PR DESCRIPTION
**Pull request by truhacevkir**

### CODE REVIEW

File: tickets/src/main/java/no/unit/nva/publication/ticket/create/TicketResolver.java [hunk 1]
```diff
@@ -232,14 +224,18 @@ private Publication toPublicationWithPublishedFiles(Publication publication) {
...
-    private AssociatedArtifact updateFileToPublished(AssociatedArtifact artifact) {
-        if (isNotAdministrativeAgreement(artifact)) {
-            return ((File) artifact).toPublishedFile();
+    //TODO: Remove unpublishable file and logic related to it after we have migrated files
+    private AssociatedArtifact openFiles(AssociatedArtifact artifact) {
+        if (artifact instanceof PendingFile<?> file) {
+            return (AssociatedArtifact) file.approve();
+        }
+        if (artifact instanceof AdministrativeAgreement administrativeAgreement) {
+            return administrativeAgreement;
```
Comment (truhacevkir) [inline tickets/src/main/java/no/unit/nva/publication/ticket/create/TicketResolver.java:236]: Handling AdministrativeAgreement in own if statement, should be removed after migration.

File: tickets/src/main/java/no/unit/nva/publication/ticket/update/UpdateTicketHandler.java [hunk 2]
```diff
@@ -213,25 +212,25 @@ private void updateUnpublishedFilesToUnpublishable(SortableIdentifier publicatio
...
-        return switch (associatedArtifact) {
-            case UnpublishedFile unpublishedFile -> unpublishedFile.toUnpublishableFile();
-            case PendingInternalFile pendingInternalFile -> pendingInternalFile.toRejectedFile();
-            case PendingOpenFile pendingOpenFile -> pendingOpenFile.toRejectedFile();
-            case null, default -> associatedArtifact;
-        };
+    //TODO: Remove unpublishable file and logic related to it after we have migrated files
+    private static AssociatedArtifact rejectFile(AssociatedArtifact associatedArtifact) {
+        if (associatedArtifact instanceof UnpublishedFile unpublishedFile) {
+            return unpublishedFile.toUnpublishableFile();
```
Comment (truhacevkir) [inline tickets/src/main/java/no/unit/nva/publication/ticket/update/UpdateTicketHandler.java:230]: Handling rejected UnpublishedFile in own if statement, should be removed after migration.

File: tickets/src/main/java/no/unit/nva/publication/ticket/update/UpdateTicketHandler.java [hunk 3]
```diff
@@ -213,25 +212,25 @@ private void updateUnpublishedFilesToUnpublishable(SortableIdentifier publicatio
...
+        } else {
+            return associatedArtifact instanceof PendingFile<?> pendingFile ? pendingFile.reject() : associatedArtifact;
```
Comment (truhacevkir) [inline tickets/src/main/java/no/unit/nva/publication/ticket/update/UpdateTicketHandler.java:232]: Is necessary cause AssociatedArtifact can be an AssociatedLink.

File: tickets/src/main/java/no/unit/nva/publication/ticket/create/TicketResolver.java [hunk 4]
```diff
@@ -232,14 +224,18 @@ private Publication toPublicationWithPublishedFiles(Publication publication) {
...
         } else {
-            return artifact;
+            return artifact instanceof File file ? file.toPublishedFile() : artifact;
```
Comment (truhacevkir) [inline tickets/src/main/java/no/unit/nva/publication/ticket/create/TicketResolver.java:238]: Is necessary cause AssociatedArtifact can be an AssociatedLink.

**Review by ketilaa — APPROVED** [review]

File: tickets/src/main/java/no/unit/nva/publication/ticket/create/TicketResolver.java (hunk 4 above)
Comment (torbjokv) [reply tickets/src/main/java/no/unit/nva/publication/ticket/create/TicketResolver.java:238]: Rejected should also be handled here maybe? What will happen with existing Open and Internal files?

Comment (truhacevkir) [reply tickets/src/main/java/no/unit/nva/publication/ticket/create/TicketResolver.java:238]: Riktig ja, godt poeng.

**Review by torbjokv — APPROVED** [review]

**Pull request merged**